src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -4781,6 +4877,19 @@ private static string MatchCharacterClass(RegexOptions options, string chExpr, s
                     $"({range0Clause} | {range1Clause})";
             }
 
+            const string Base = nameof(Base);
+            if (!requiredHelpers.ContainsKey(Base))
+            {
+                requiredHelpers.Add(Base, new string[]
+                {
+                    $"internal class {Base} : RegexRunner",
+                    $"{{",
+                    $"    /// <summary>Determines whether the specified character in is in the specified character class.</summary>",
+                    $"    internal static new bool CharInClass(char ch, string charClass) => RegexRunner.CharInClass(ch, charClass);",
```

Comment:
Out of curiosity, what's the benefit of having this class instead of just calling into the helper directly?

Reply:
> what's the benefit of having this class instead of just calling into the helper directly?

It compiles :)

RegexRunner.CharInClass is unfortunately protected rather than public (I plan to open an issue to change its visiblity).

Reply:
Ah, that makes sense, I didn't realize previous usages were never in the utilities helper